manim/scene/scene.py

```diff
@@ -903,6 +771,81 @@ def wait_until(self, stop_condition, max_time=60):
         """
         self.wait(max_time, stop_condition=stop_condition)
 
+    def play_internal(self, *args, **kwargs):
+        """
+        This method is used to prep the animations for rendering,
+        apply the arguments and parameters required to them,
+        render them, and write them to the video file.
+
+        Parameters
+        ----------
+        *args : Animation or mobject with mobject method and params
+        **kwargs : named parameters affecting what was passed in *args e.g
+            run_time, lag_ratio etc.
+        """
+        if len(args) == 0:
+            warnings.warn("Called Scene.play with no animations")
+            return
+
+        animations = self.compile_play_args_to_animation_list(*args, **kwargs)
+        if (
+            len(animations) == 1
+            and isinstance(animations[0], Wait)
+            and not self.should_update_mobjects()
```

Comment:
**Question** : what will happen if there are several animations including a `Wait` and mobjects does not need to be updated ?

Reply:
The rendered animation would still be the same, but the rendering process wouldn't benefit from this optimization. That said, that'd be a rather odd way to write a scene.